### PR TITLE
[metricbeat] [gcp] refactor metric filter builder

### PR DESCRIPTION
**Pull request by ghost**
## What does this PR do?

Extracts the logic to build metrics filters into a function to avoid redundant code.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

**Review by endorama — APPROVED** [review]
~I'm approving this but I'm going to request a second review from the team.~

Just noticed https://github.com/elastic/beats/pull/33728 is not merged yet, once merged will require an update to this PR. Overall the logic looks good to me! (and tests are green)

**Comment by endorama** [comment]
https://github.com/elastic/beats/pull/33728 has been merged. @gpop63 may you rebase this?

File: x-pack/metricbeat/module/gcp/metrics/metrics_requester.go
Comment (kaiyan-sheng) [inline x-pack/metricbeat/module/gcp/metrics/metrics_requester.go:172]: nit: with directly `return f`, we missed the `r.logger.Debugf("ListTimeSeries API filter = %s", f)` debug logging here.

Comment (endorama) [reply x-pack/metricbeat/module/gcp/metrics/metrics_requester.go]: I slightly refactored the code so that it always log that line in https://github.com/elastic/beats/pull/33560/commits/8549f0bff3964e012a7b3f56b2ea884ede2bdd71

What do you think?

File: x-pack/metricbeat/module/gcp/metrics/metrics_requester_test.go
Comment (aspacca) [inline x-pack/metricbeat/module/gcp/metrics/metrics_requester_test.go:88]: can we increase coverage in order to include all the scenarios that the refactored method can produce?

File: x-pack/metricbeat/module/gcp/metrics/metrics_requester_test.go
Comment (endorama) [inline x-pack/metricbeat/module/gcp/metrics/metrics_requester_test.go:94]: Here `gcp.ServiceCompute` is used as an example for the overall behaviour. All specific behaviours are tested in isolation in other functions:
- `TestBuildRegionsFilter` verifies that the region based filter creation works as intended
- `TestIsAGlobalService` verifies each service (and default) corresponding "global" status flag value
- `TestGetServiceLabelFor` verifies each service (and default) corresponding label value
- `TestTrimWildcard` verifies the wildcard remover produces correct results
- `TestBuildLocationFilter` verifies the location based filter creation works as intended

Keeping tests `TestIsAGlobalService` and `TestGetServiceLabelFor` up to date with the available services and labels in the `gcp` package will ensure all combinations to be covered.

**Review by francescayeye — APPROVED** [review]

**Pull request merged**
